spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/loadbalancer/core/HealthCheckServiceInstanceListSupplier.java

```diff
@@ -74,32 +77,33 @@ public HealthCheckServiceInstanceListSupplier(ServiceInstanceListSupplier delega
 	}
 
 	private void initInstances() {
-		delegate.get().subscribe(delegateInstances -> {
-			instances.clear();
-			instances.addAll(delegateInstances);
-		});
-
-		Flux<List<ServiceInstance>> healthCheckFlux = healthCheckFlux();
-
-		healthCheckFlux.subscribe(verifiedInstances -> {
-			healthyInstances.clear();
-			healthyInstances.addAll(verifiedInstances);
-		});
+		healthCheckDisposable = delegate.get().doOnNext(delegateInstances -> {
```

Comment:
I do no see contract about `ServiceInstanceListSupplier#get()`   
`Flux` does it considered to be finite?

Reply:
yes

Reply:
If it is finite (and assume that it has only one emit) what is the reason to return `Flux` instead of `Mono` from  `ServiceInstanceListSupplier#get`?  
If there is such contract maybe it is a good idea to document it, because it is not clear from current returning type